frontend/packages/pipelines-plugin/src/components/pipelineruns/PipelineRunDetailsPage.tsx

```diff
@@ -16,7 +16,7 @@ const PipelineRunDetailsPage: React.FC<DetailsPageProps> = (props) => {
   const { t } = useTranslation();
   const { kindObj, match } = props;
   const menuActions: KebabAction[] = useMenuActionsWithUserAnnotation(
-    getPipelineRunKebabActions(true),
+    getPipelineRunKebabActions(t)(true),
```

Comment:
I know this is something other people are doing a lot in console, but I always get dinged for this whenever Sam reviews my PRs. I think he may have concerns about passing TFunctions around.

Reply:
I'm not a fan of it either... but we chose a hook approach in the Console. Utilities have words that need translated, they cannot use the hook. I'm open to hearing about alternatives... but unless we want to just refresh the browser, we can't really go with the static approach `i18next.t(...)`.

In some of our static models, we went with the `key` approach; Kebabs and K8sKinds... but that only gets us so far with infrastructure. We'd need to start baking into common components an approach that uses keys instead. In the above case, it's an error modal I believe. I suppose we could instead of taking a string, take a "stringKey". 

This harks back to my comment on Slack about adopting i18n fully as a first class citizen. Display strings should not be allowed anymore imo, we force the hand of the developer into i18n... but obv that only works in most cases as Sam noted the exception for k8sKinds we discover through api fetching.

Reply:
Slack thread happened here: https://coreos.slack.com/archives/C0176H51F8V/p1618412328043000

Essentially the answer is to use `i18n.t` when inside functions as it gets re-executed each time the function is called and thus pulls the latest setting of the language at that time.